source/main/physics/collision/Collisions.cpp

```diff
@@ -1353,6 +1304,9 @@ void primitiveCollision(node_t *node, Vector3 &force, const Vector3 &velocity, c
                 Freaction += -Vnormal * node->mass / dt; // Newton's second law
             }
             if (Freaction < 0) Freaction = 0.0f;
+            // We only update those if we handle ground collisions

```

Comment:
I had a hard time deciding whether or not to clear the variables. But my conclusion was that it is clearer this way. Since the comments already say:

`last .. slip`, `last .. ground model` and not `current .. slip`, `current .. ground model`.

But I guess you're right, we should just rename them accordingly.

Reply:
The underlying idea was to later on expand the `nd_collision_gm` and `nd_collision_slip` logic to inter truck collisions. And if we do that we can't just clear them in `CalcNodes()`.